### ISSUE TITLE
Customer is subscribed to VPN on a deleted account

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#13462]
**TL;DR**: Since we make a bunch of successive API calls to the auth server during checkout, it’s possible for customers to get into partially processed account/subscription states. We should move this work into the auth server, so it cannot be interrupted.

- - -
During triage today, we noticed an `invoice.payment_failed` [webhook](https://dashboard.stripe.com/events/evt_1LEAp8JNcmPzuWtRPMqlwowW) that permafailed. Upon investigating, we found that this email address has two Stripe customers (with the same CC) with two different uids created on the same day within minutes of each other:

1. [https://dashboard.stripe.com/customers/cus_KzkK0clOMdLSQR](https://dashboard.stripe.com/customers/cus_KzkK0clOMdLSQR|smart-link)  was created first at 12:32pm PST January 19th.
  1. This customer had an active VPN subscription that lapsed due to failed payments (causing the `invoice.payment_failed` webhook above).
  1. The uid for this customer is not found in our database per the admin panel.
1. [https://dashboard.stripe.com/customers/cus_KzkVRO6WhYjMld](https://dashboard.stripe.com/customers/cus_KzkVRO6WhYjMld|smart-link) was created second at 12:44pm PST January 19th.
  1. This customer has no subscription history.
  1. The uid for this customer IS in our database per the admin panel.

The consequence for this user is that they are being charged for a subscription (#1) that they can’t use, since they can’t log in on account #1. They can’t submit a support ticket through either account via Subscription Management, since they can’t log into #1 and don’t have an active subscription for #2.

In the short term, we should contact this user to either a) transfer their subscription to account #2 or b) refund them the entire amount they’ve been charged for #1. If they don’t respond, we should refund the card on file.

In the long term, we likely want to:

* Update our support playbook to include a check if a user says they’re being charged but doesn’t have an active subscription to see if there are multiple accounts/uids for that email address and perform the steps described above to transfer or refund.
* Move all of the successive API calls from the payments server to the auth server to ensure we don’t get into incomplete processing states e.g. on page navigation/refresh.
  * [~accountid:5e3400fb6bd5c40ca395d189] Has some ideas on what this could look like, but it would be a sizeable refactor.

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/FXA-5442)

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

{quote}In the short term, we should contact this user to either a) transfer their subscription to account #2 or b) refund them the entire amount they’ve been charged for #1. If they don’t respond, we should refund the card on file.{quote}

Joseph Anthony Cuevas : Can you help us with the short-term fix? FYI the long term fix will likely involve a Support process change, which we will follow up with you on later.

Anna Sobiepanek , Chelsea Lewis : Since it’s after the work day, I’m pinging you here to see if I should drop my current tickets for the sprint and pick this up? This is 5 story points, so I want to call out the risk that this may be started but not completed by the end of the maintenance sprint, given the Wellness Day and holiday in the U.S.

It’d be tricky to determine how many users this affects, but for those it does, the issue is pretty severe. Unfortunately I think this bug has existed for a while. I will try to dig up some prior bugs where we have seen similar consequences and link them here. I think those fixes landed after this customer subscribed, so it’s possible the risk of this issue occuring in production today is reduced.

**Comment by data-sync-user** [comment]
➤ Joseph Anthony Cuevas commented:

Bianca Danforth The customer has been refunded and notified. Please let me know if you need anything else.

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Thanks Joe; Benjamin Bangert : I gave this ticket 5 points – but I’m wondering if it is actually more than that based on our conversation yesterday.

I’ll make this a spike, since we should be explicit what all needs to change first.

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Per Ben:

If this takes 2-3 weeks of active investigation to fix this for 20 customers, that’s a lot of engineering work. We know it affects a very small number of people.

Let’s go with a two-pointer short-term solution (described here ( https://mozilla-hub.atlassian.net/browse/FXA-5442?focusedCommentId=562858 )) for now.
